src/v/http/client.cc

```diff
@@ -302,10 +302,10 @@ ss::future<iobuf> client::response_stream::recv_some() {
               // we'll have to manually check the headers and the last chunk
               // status.
               boost::beast::error_code ec;
-              if (!_parser.is_header_done()) {
-                  // We can't put EOF before all header bytes are received
+              if (!_parser.got_some()) {
```

Comment:
what if we disconnected before all headers are received?

Reply:
Boost parser has the following states (listing only first 3): nothing_yet = 0, start_line = 1, fields = 2.

is_header_done is defined as `state_ > state::fields`, negating it becomes state_ <= state::fields

put_eof [starts like this](https://github.com/boostorg/beast/blob/42b6387ae45481cad4ef7134a57affb637ffcab5/include/boost/beast/http/impl/basic_parser.ipp#L238-L244):

```cpp
BOOST_ASSERT(got_some());
if( state_ == state::start_line ||
        state_ == state::fields)
{
    BOOST_BEAST_ASSIGN_EC(ec, error::partial_message);
    return;
}
```

We need to cover only got_some which is defined as `state_ != nothing_yet`. Which I did.

Reply:
OK, makes sense